pkg/infra/proposer.go

```diff
@@ -136,6 +137,8 @@ func (b *Broadcaster) Start(envs <-chan *Elements, errorCh chan error, done <-ch
 
 		case <-done:
```

Comment:
do we still need `done` chan?

Reply:
We no longer need `done` chan. We should use `ctx.Done()` instead of `done` chan.
I will refactor these code. Please review the code for me again.